datadog-lambda.gemspec

```diff
@@ -31,6 +31,7 @@ Gem::Specification.new do |spec|
   spec.require_paths = ['lib']
 
   spec.add_dependency 'aws-xray-sdk', '~> 0.11.3'
+  spec.add_dependency 'dogstatsd-ruby', '~> 5.6'
```

Comment:
Just out of curiosity, does the gem `datadog-lambda` get included in the user's Gemfile (e.g. we ask them to add `gem 'datadog-lambda'` to their Gemfile)?

I ask because, if we are added to their Gemfile, we normally try to be nice neighbors and not create dependency requirements that are too strict for their application.

For example, here we are requiring that they have `dogstatsd-ruby` >= 5.6 and < 6. If they have 5.5, `datadog-lambda` will complain. If there's no strong reason to limit this to >= 5.5, I suggest dropping this requirement down to `~> 5` (which is the same as `~> 5.0`). You even go `~> 4` if you are feeling, but that's probably not required, since the last 4.x is over two years old.
< 6 is a totally valid restriction, so I think that's a good call.

BUT, if we are never in the user's Gemfile, then forget everything I said above, and keep it like it is today; if we are not in the user's Gemfile, we have the `datadog-lambda.gemspec` all for our own use only, and we have full control of all dependencies.

Reply:
Thanks @marcotc,

Yes, we actually are included in their Gemfile if they use a Dockerfile or custom instrumentation.
Will proceed by lowering this requirement to `~> 5.0`